### DISCUSSION TITLE
Sometimes phase 2 takes rougly twice as long to complete - adding sustantial time

### DISCUSSION THREAD

**Discussion by Ron-ski** [madMAx43v3r/chia-plotter#1004] [General]
I and others have found a tendency for K33 and K34 plots to take much longer in phase two. If you're doing more than one plot, and the first plot phase 2 takes a longer time (twice as long), then all plots in that run will take longer.

If you stop the run, and start again then it may go fast or slow, we may have to stop several times until it deicides to go fast.

For full info see this thread [Sometime plots take 10 minutes longer to complete](https://chiaforum.com/t/sometime-plots-take-10-minutes-longer-to-complete/15794)
 
I think this may have affected k32 as well, but at the moment I'm not certain - I will need to test.

66 minute plot

Crafting plot 4 out of 27 (2022/03/05 14:30:30)
Process ID: 12081
Number of Threads: 18
Number of Buckets P1:    2^8 (256)
Number of Buckets P3+P4: 2^7 (128)
Pool Puzzle Hash:  <snip>
Farmer Public Key: <snip>
Working Directory:   /media/chiamining/Raid0/
Working Directory 2: /mnt/ramdisk/
Plot Name: plot-k33-2022-03-05-14-30-eb1b093c53fb8be13f943506f711b17b0ab4f262786e1e90e746978c8e91fcd5
[P1] Table 1 took 35.7205 sec
[P1] Table 2 took 201.269 sec, found 8589864034 matches
[P1] Table 3 took 259.654 sec, found 8589840559 matches
[P1] Table 4 took 290.663 sec, found 8589779461 matches
[P1] Table 5 took 286.452 sec, found 8589676618 matches
[P1] Table 6 took 259.396 sec, found 8589509516 matches
[P1] Table 7 took 233.741 sec, found 8589166604 matches
Phase 1 took 1566.91 sec
[P2] max_table_size = 8589934592
[P2] Table 7 scan took 45.7622 sec
[P2] Table 7 rewrite took 78.1122 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 54.534 sec
[P2] Table 6 rewrite took 100.178 sec, dropped 1162575443 entries (13.5348 %)
[P2] Table 5 scan took 63.5149 sec
[P2] Table 5 rewrite took 79.6154 sec, dropped 1523945922 entries (17.7416 %)
[P2] Table 4 scan took 65.0739 sec
[P2] Table 4 rewrite took 78.0684 sec, dropped 1657688927 entries (19.2984 %)
[P2] Table 3 scan took 69.7358 sec
[P2] Table 3 rewrite took 77.5137 sec, dropped 1710077829 entries (19.9081 %)
[P2] Table 2 scan took 66.6757 sec
[P2] Table 2 rewrite took 75.665 sec, dropped 1731083900 entries (20.1526 %)
Phase 2 took 887.372 sec
Wrote plot header with 252 bytes
[P3-1] Table 2 took 89.4178 sec, wrote 6858780134 right entries
[P3-2] Table 2 took 80.1316 sec, wrote 6858780134 left entries, 6858780134 final
[P3-1] Table 3 took 136.35 sec, wrote 6879762730 right entries
[P3-2] Table 3 took 86.6875 sec, wrote 6879762730 left entries, 6879762730 final
[P3-1] Table 4 took 138.419 sec, wrote 6932090534 right entries
[P3-2] Table 4 took 87.0621 sec, wrote 6932090534 left entries, 6932090534 final
[P3-1] Table 5 took 135.874 sec, wrote 7065730696 right entries
Copy to /media/chiamining/DiskBay6SAS10TB/ChiaNFTPlots/plot-k33-2022-03-05-13-24-161f3dd250b1dcb3e49215a91285689472cbed96ee38a72fb187bddd5d023bf9.plot finished, took 3277.77 sec, 65.2344 MB/s avg.
[P3-2] Table 5 took 88.7771 sec, wrote 7065730696 left entries, 7065730696 final
[P3-1] Table 6 took 138.697 sec, wrote 7426934073 right entries
[P3-2] Table 6 took 91.9238 sec, wrote 7426934073 left entries, 7426934073 final
[P3-1] Table 7 took 162.496 sec, wrote 8589166604 right entries
[P3-2] Table 7 took 107.125 sec, wrote 8589166604 left entries, 8589166604 final
Phase 3 took 1353.5 sec, wrote 43752464771 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 149.794 sec, final plot size is 224225084917 bytes
Total plot creation time was 3957.69 sec (65.9614 min)

74 minute plot

Crafting plot 1 out of 29 (2022/03/05 00:30:37)
Process ID: 2576
Number of Threads: 18
Number of Buckets P1:    2^8 (256)
Number of Buckets P3+P4: 2^7 (128)
Pool Puzzle Hash:  <snip>
Farmer Public Key: <snip>
Working Directory:   /media/chiamining/Raid0/
Working Directory 2: /mnt/ramdisk/
Plot Name: plot-k33-2022-03-05-00-30-83028f34f0b27c67fdb82f07a98bceef563c6329a5efc12b5131f1777ac199b4
[P1] Table 1 took 34.384 sec
[P1] Table 2 took 194.439 sec, found 8589850456 matches
[P1] Table 3 took 253.352 sec, found 8589867008 matches
[P1] Table 4 took 283.601 sec, found 8589849728 matches
[P1] Table 5 took 278.964 sec, found 8589799812 matches
[P1] Table 6 took 250.026 sec, found 8589694079 matches
[P1] Table 7 took 228.02 sec, found 8589342556 matches
Phase 1 took 1522.8 sec
[P2] max_table_size = 8589934592
[P2] Table 7 scan took 43.353 sec
[P2] Table 7 rewrite took 77.0855 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 40.8711 sec
[P2] Table 6 rewrite took 205.402 sec, dropped 1162536740 entries (13.5341 %)
[P2] Table 5 scan took 51.281 sec
[P2] Table 5 rewrite took 196.804 sec, dropped 1523931872 entries (17.7412 %)
[P2] Table 4 scan took 59.3791 sec
[P2] Table 4 rewrite took 183.439 sec, dropped 1657675510 entries (19.2981 %)
[P2] Table 3 scan took 61 sec
[P2] Table 3 rewrite took 232.112 sec, dropped 1710071639 entries (19.908 %)
[P2] Table 2 scan took 59.3086 sec
[P2] Table 2 rewrite took 184.702 sec, dropped 1731047063 entries (20.1522 %)
Phase 2 took 1426.65 sec
Wrote plot header with 252 bytes
[P3-1] Table 2 took 88.2685 sec, wrote 6858803393 right entries
[P3-2] Table 2 took 79.7635 sec, wrote 6858803393 left entries, 6858803393 final
[P3-1] Table 3 took 133.444 sec, wrote 6879795369 right entries
[P3-2] Table 3 took 84.5177 sec, wrote 6879795369 left entries, 6879795369 final
[P3-1] Table 4 took 136.256 sec, wrote 6932174218 right entries
[P3-2] Table 4 took 85.6728 sec, wrote 6932174218 left entries, 6932174218 final
[P3-1] Table 5 took 137.01 sec, wrote 7065867940 right entries
[P3-2] Table 5 took 86.9061 sec, wrote 7065867940 left entries, 7065867940 final
[P3-1] Table 6 took 141.836 sec, wrote 7427157339 right entries
[P3-2] Table 6 took 92.2599 sec, wrote 7427157339 left entries, 7427157339 final
[P3-1] Table 7 took 158.31 sec, wrote 8589342556 right entries
[P3-2] Table 7 took 105.892 sec, wrote 8589342556 left entries, 8589342556 final
Phase 3 took 1340.45 sec, wrote 43753140815 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 148.836 sec, final plot size is 224228717351 bytes
Total plot creation time was 4438.85 sec (73.9809 min)

**Comment by mech2you** [comment]
I have the same problem with k33 plots. 10 oder more different between Plots. When i restart the plotter sometimes it finsh the 72 Plots in 77min and when the error ist happend it for every of the 72 Plots 94min...

Windows 10 21H1 19043.1620 AMD 5950x 64GB 3600 2TB AN1500

**Comment by Grosik1577** [comment]
AMD 5950x 128GB 3600 2x1.6TB PM1735 Clear Linux kernel 5.12
I have the same problem with k33 and k34 plots. As i observed the main difference starts from
[P2] Table 6 rewrite
If it goes SLOW than, as per Htop,  all  [P2] Table x rewrites are Single Core 100% and other cores 10-20%
[P2] max_table_size = 8591960036
[P2] Table 7 scan took 37.1078 sec
[P2] Table 7 rewrite took 62.7559 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 43.7843 sec
[P2] Table 6 rewrite took 346.423 sec, dropped 1162409851 entries (13.5306 %)
[P2] Table 5 scan took 42.7258 sec
[P2] Table 5 rewrite took 317.952 sec, dropped 1523678697 entries (17.737 %)
[P2] Table 4 scan took 43.0958 sec
[P2] Table 4 rewrite took 315.809 sec, dropped 1657525624 entries (19.2956 %)
[P2] Table 3 scan took 41.9618 sec
[P2] Table 3 rewrite took 320.874 sec, dropped 1710002810 entries (19.9067 %)
[P2] Table 2 scan took 41.8325 sec
[P2] Table 2 rewrite took 320.281 sec, dropped 1731051201 entries (20.1521 %)
Phase 2 took 1951.04 sec
If it goes FAST than all cores are 70-90% used
[P2] max_table_size = 8591485063
[P2] Table 7 scan took 27.2468 sec
[P2] Table 7 rewrite took 56.8793 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 45.1812 sec
[P2] Table 6 rewrite took 90.4128 sec, dropped 1162401340 entries (13.5309 %)
[P2] Table 5 scan took 43.7353 sec
[P2] Table 5 rewrite took 86.8796 sec, dropped 1523755776 entries (17.7379 %)
[P2] Table 4 scan took 42.9656 sec
[P2] Table 4 rewrite took 85.4762 sec, dropped 1657598359 entries (19.2965 %)
[P2] Table 3 scan took 42.7092 sec
[P2] Table 3 rewrite took 84.9902 sec, dropped 1710037887 entries (19.9072 %)
[P2] Table 2 scan took 42.5686 sec
[P2] Table 2 rewrite took 84.6988 sec, dropped 1731088017 entries (20.1523 %)
Phase 2 took 750.125 sec
per final i have difference
Total plot creation time was 4306.52 sec (71.7754 min)
and
Total plot creation time was 2983.01 sec (49.7169 min)
Hope it helps to find the problem.
Best regards

**Comment by mech2you** [comment]
Same settings, same maschine... 14 min difference... I thing in phase 2 is the error... Tabele rewrite needs double so long as normal
<img width="988" alt="K33 80Min" src="https://user-images.githubusercontent.com/61209759/162647043-e098a217-7fa6-4bad-bc80-5cc49f79f9d2.png">
<img width="751" alt="K33 94Min" src="https://user-images.githubusercontent.com/61209759/162647045-8ffda74f-a3c8-4240-8bca-5b69322366fc.png">

**Comment by bladeuserpi** [comment]
```
I am also seeing that "random-but-sticky-bad-phase2-performance" effect for K34 runs:
- in the same run performance can vary a bit, e.g. delta~10-30 minutes  (see below)
- When you see "phase2 slow run" you can press ctrl-c and restart it (without reboot!) until you see better performance
- K34: Good performance is 100-111 minutes, bad performance is 123-152 minutes
- machine is 2x E5-2699v4, 512 GB

Slow phase2:
mm81466.19007.out:Total plot creation time was 8080.94 sec (134.682 min)
mm81466.19007.out:Total plot creation time was 8168.16 sec (136.136 min)
mm81466.19007.out:Total plot creation time was 8143.31 sec (135.722 min)
mm81466.19007.out:Total plot creation time was 8948.08 sec (149.135 min)
mm81466.19007.out:Total plot creation time was 8762.63 sec (146.044 min)
mm81466.19007.out:Total plot creation time was 7389.87 sec (123.165 min)
mm81466.19007.out:Total plot creation time was 9173.63 sec (152.894 min) -> Worst RUN so far

Quick Phase2:
mm114205.25584.out:Total plot creation time was 6705.97 sec (111.766 min)
mm114205.25584.out:Total plot creation time was 6283.51 sec (104.725 min)
mm114205.25584.out:Total plot creation time was 6256.2 sec (104.27 min)
mm114205.25584.out:Total plot creation time was 6402.21 sec (106.704 min)
mm114205.25584.out:Total plot creation time was 6087.02 sec (101.45 min)
mm114205.25584.out:Total plot creation time was 6042.22 sec (100.704 min)  --> Best RUN so far

**Reply by bladeuserpi** [reply]
```
This is runtime per phase:

SLOW phase2 
# grep -e took -e Total mm81466.19007.out  |grep -v Table |cat -n
     1	Phase 1 took 2883.25 sec
     2	Phase 2 took 2917.1 sec
     3	Phase 3 took 2071.66 sec, wrote 87494746318 entries to final plot
     4	Phase 4 took 208.777 sec, final plot size is 461469953942 bytes
     5	Total plot creation time was 8080.94 sec (134.682 min)
     6	Phase 1 took 2568.37 sec
     7	Phase 2 took 3313.88 sec
     8	Phase 3 took 2062.84 sec, wrote 87515445258 entries to final plot
     9	Phase 4 took 222.847 sec, final plot size is 461601679224 bytes
    10	Total plot creation time was 8168.16 sec (136.136 min)
    11	Phase 1 took 2639 sec
    12	Phase 2 took 3241.84 sec
    13	Phase 3 took 2042.21 sec, wrote 87496893140 entries to final plot
    14	Phase 4 took 220.033 sec, final plot size is 461482658650 bytes
    15	Total plot creation time was 8143.31 sec (135.722 min)
    16	Phase 1 took 2562.06 sec
    17	Phase 2 took 4069.53 sec
    18	Phase 3 took 2094.65 sec, wrote 87506459463 entries to final plot
    19	Phase 4 took 221.486 sec, final plot size is 461545006120 bytes
    20	Total plot creation time was 8948.08 sec (149.135 min)
    21	Phase 1 took 2611.7 sec
    22	Phase 2 took 3871.15 sec
    23	Phase 3 took 2054.92 sec, wrote 87505559429 entries to final plot
    24	Phase 4 took 224.642 sec, final plot size is 461539277464 bytes
    25	Total plot creation time was 8762.63 sec (146.044 min)
    26	Phase 1 took 2526.57 sec
    27	Phase 2 took 2601.14 sec
    28	Phase 3 took 2040.63 sec, wrote 87510372003 entries to final plot
    29	Phase 4 took 221.329 sec, final plot size is 461569419281 bytes
    30	Total plot creation time was 7389.87 sec (123.165 min)
    31	Phase 1 took 2547.22 sec
    32	Phase 2 took 4339.63 sec
    33	Phase 3 took 2064.92 sec, wrote 87519247730 entries to final plot
    34	Phase 4 took 221.643 sec, final plot size is 461626939372 bytes
    35	Total plot creation time was 9173.63 sec (152.894 min)

QUICK phase2
# grep -e took -e Total mm114205.25584.out  |grep -v Table |cat -n
     1	Phase 1 took 3142.75 sec
     2	Phase 2 took 1228.35 sec
     3	Phase 3 took 2104.64 sec, wrote 87500926404 entries to final plot
     4	Phase 4 took 230.065 sec, final plot size is 461508876009 bytes
     5	Total plot creation time was 6705.97 sec (111.766 min)
     6	Phase 1 took 2680.14 sec
     7	Phase 2 took 1239.32 sec
     8	Phase 3 took 2135.25 sec, wrote 87521143937 entries to final plot
     9	Phase 4 took 228.56 sec, final plot size is 461638893970 bytes
    10	Total plot creation time was 6283.51 sec (104.725 min)
    11	Phase 1 took 2684.18 sec
    12	Phase 2 took 1239.2 sec
    13	Phase 3 took 2122.89 sec, wrote 87495948162 entries to final plot
    14	Phase 4 took 209.737 sec, final plot size is 461477407694 bytes
    15	Total plot creation time was 6256.2 sec (104.27 min)
    16	Phase 1 took 2736.48 sec
    17	Phase 2 took 1315.42 sec
    18	Phase 3 took 2122.62 sec, wrote 87514635432 entries to final plot
    19	Phase 4 took 227.468 sec, final plot size is 461597137700 bytes
    20	Total plot creation time was 6402.21 sec (106.704 min)
    21	Phase 1 took 2558.55 sec
    22	Phase 2 took 1231.01 sec
    23	Phase 3 took 2072.67 sec, wrote 87522404019 entries to final plot
    24	Phase 4 took 224.561 sec, final plot size is 461646963044 bytes
    25	Total plot creation time was 6087.02 sec (101.45 min)
    26	Phase 1 took 2538.58 sec
    27	Phase 2 took 1240.4 sec
    28	Phase 3 took 2049.24 sec, wrote 87529817262 entries to final plot
    29	Phase 4 took 213.782 sec, final plot size is 461694095744 bytes
    30	Total plot creation time was 6042.22 sec (100.704 min)

**Comment by Ron-ski** [comment]
Set my plotter doing a batch of k33s this morning, just came home to find its in go slow mode, so 70 minute per plot instead of 56 minutes per plot  :-(
